src/main/java/com/serjltt/moshi/adapters/DeserializeOnly.java

```diff
@@ -0,0 +1,26 @@
+package com.serjltt.moshi.adapters;
+
+import com.squareup.moshi.JsonQualifier;
+import com.squareup.moshi.Moshi;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Indicates that the annotated field may only be deserialized.
+ *
+ * <p>To leverage from {@linkplain DeserializeOnly} the
+ * {@linkplain DeserializeOnlyJsonAdapter#FACTORY} must be added to a
+ * {@linkplain Moshi Moshi instance}:
+ *
+ * <pre><code>
+ *   Moshi moshi = new Moshi.Builder()
+ *      .add(DeserializeOnlyJsonAdapter.FACTORY)
+ *      .build();
+ * </code></pre>
+ */
+@Documented
+@JsonQualifier
+@Retention(RetentionPolicy.RUNTIME)
```

Comment:
Yeah I didn't add those since the other annotations don't declare them as well we might want to add them to every annotation